cmd/contour/gatewayprovisioner.go

```diff
@@ -52,6 +53,10 @@ func registerGatewayProvisioner(app *kingpin.Application) (*kingpin.CmdClause, *
 		Default(config.metricsBindAddress).
 		StringVar(&config.metricsBindAddress)
 
+	cmd.Flag("leader-election-namespace", "The namespace in which the leader election resource will be created.").
+		Default(config.leaderElectionNamespace).
```

Comment:
I'd prefer to use the Downward API to set an env var containing the namespace that the gateway provisioner is deployed in, and then use the value of that env var as the default here. That way if the user deploys the gateway provisioner into a different namespace, they don't also need to set this flag. This is how Contour itself does it -- see https://github.com/projectcontour/contour/blob/main/examples/contour/03-contour.yaml#L77-L82 for defining the env var using the downward API, and then https://github.com/projectcontour/contour/blob/a20cac5011b5c2a3614b542dd988a6fb5369d625/cmd/contour/serve.go#L130 for how it's used as the default.

Reply:
yeah thats a good point, was thinking we might do that in a follow up (previous changes in the PR had something similar) but might as well do it here

maybe use `CONTOUR_PROVISIONER_NAMESPACE` or something similar as the env var

Reply:
done